cpp/tests/filter/filter_test.cpp

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/random.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/detail/iterator.cuh>
+#include <cudf/filter.hpp>
+#include <cudf/jit/runtime_support.hpp>
+
+namespace filters {
+
+struct TypeSupportTest : public cudf::test::BaseFixture {
+ protected:
+  void SetUp() override
+  {
+    if (!cudf::is_runtime_jit_supported()) {
+      GTEST_SKIP() << "Skipping tests that require runtime JIT support";
+    }
+  }
+
+  static constexpr char const* udf =
+    R"***(
+    template<typename T>
+    __device__ void is_equal(bool * out, T a, T b) { *out = (a == b); }
+    )***";
+};
+
+template <typename T>
+struct NumericSupportTest : public TypeSupportTest {};
+
+using NumericTypesNotBool =
+  cudf::test::Concat<cudf::test::IntegralTypesNotBool, cudf::test::FloatingPointTypes>;
+
+TYPED_TEST_SUITE(NumericSupportTest, NumericTypesNotBool);
+
+TYPED_TEST(NumericSupportTest, Assertions)
```

Comment:
```suggestion
TYPED_TEST(FilterNumericTest, Basic)
```
The `Assertions` title feels like this should be testing errors or asserts.

Reply:
Yes, I understand (then it should be NoAsssertions). I still think we should have a `Filter-Test` prefix. I can help later when trying to test a subset of the tests using a wildcard.